tests/appsec/waf/test_addresses.py

```diff
@@ -97,16 +97,20 @@ def test_specific_key(self):
     def setup_specific_key2(self):
         self.r_sk_4 = weblog.get("/waf/", headers={"X_Filename": "routing.yml"})
 
-    @missing_feature(library="python")
     @irrelevant(library="ruby", reason="Rack transforms underscores into dashes")
     @irrelevant(library="php", reason="PHP normalizes into dashes; additionally, matching on keys is not supported")
     @missing_feature(weblog_variant="spring-boot-3-native", reason="GraalVM. Tracing support only")
     def test_specific_key2(self):
         """attacks on specific header X_Filename, and report it"""
-
-        interfaces.library.assert_waf_attack(
-            self.r_sk_4, pattern="routing.yml", address="server.request.headers.no_cookies", key_path=["x_filename"]
-        )
+        try:
+            interfaces.library.assert_waf_attack(
+                self.r_sk_4, pattern="routing.yml", address="server.request.headers.no_cookies", key_path=["x_filename"]
+            )
+        except ValueError:
```

Comment:
can it be an assert_waf_attack(x_filename) or assert_waf_attack(x-filename) 
instead of a try clause?

Reply:
(maybe a comment saying some frameworks normalize _ to - before it can get to the waf)

Reply:
no because if it fails, it doesn't return False but raise an Exception. So the only way is to catch it.

Reply:
Well, the try/except is indeed not the best. But adding the capacity of supporting several key_path (or a regex) to `assert_waf_attack` will over-complexify it. As it's the only use case, let's go with it. i'll reconsider and do a cleaner way if there are more use case in the futur like this